File: No_1334/Find the City With the Smallest Number of Neighbors at a Threshold Distance.py
class Solution:
    def findTheCity(self, n: int, edges: List[List[int]], distanceThreshold: int) -> int:
        dis = [[float('inf')] * n for _ in range(n)]
        
        for i, j, w in edges:
            dis[i][j] = dis[j][i] = w
            
        for i in range(n):
            dis[i][i] = 0
            
        for k in range(n):
            for i in range(n):
                for j in range(n):
                    dis[i][j] = min(dis[i][j], dis[i][k] + dis[k][j])
        
        ret = -1
        min_num_of_cities = float('inf')
        for i in reversed(range(n)):
            total = sum(1 for d in dis[i] if d <= distanceThreshold)
            if total < min_num_of_cities:
                ret = i
                min_num_of_cities = total
                
        return ret
        
        

File: No_1334/Find the City With the Smallest Number of Neighbors at a Threshold Distance.cpp
class Solution {
public:
    int findTheCity(int n, vector<vector<int>>& edges, int distanceThreshold) {
        vector<vector<long long>> dis(n, vector<long long>(n, INT_MAX));
        
        for (auto edge : edges) {
            dis[edge[0]][edge[1]] = edge[2];
            dis[edge[1]][edge[0]] = edge[2];
        }
        
        for (int i = 0; i < n; i++)
            dis[i][i] = 0;
        
        
        for (int k = 0; k < n; k++)
            for (int i = 0; i < n; i++) 
                for (int j = 0; j < n; j++)
                    dis[i][j] = min(dis[i][j], dis[i][k] + dis[k][j]);
        
        int ret = -1, minNumOfCities = INT_MAX;
        for (int i = n - 1; i >= 0; i--) {
            int total = 0;
            for (int d : dis[i]) {
                if (d <= distanceThreshold)
                    total++;
            }
            
            if (total < minNumOfCities) {
                ret = i;
                minNumOfCities = total;
            }
        }
        
        return ret;
    }
};